File: backend/app/schemas/gelbooru.py
"""Gelbooru tag search schemas."""

from pydantic import BaseModel


class GelbooruTag(BaseModel):
    id: int
    name: str
    count: int
    type: int  # 0=general, 1=artist, 3=copyright, 4=character, 5=metadata
    ambiguous: bool = False


class GelbooruTagSearchResponse(BaseModel):
    tags: list[GelbooruTag]
    query: str

File: backend-rs/src/dto/gelbooru.rs
//! Gelbooru tag search DTOs (mirror schemas/gelbooru.py).

#[derive(Debug, Clone, serde::Serialize)]
pub struct GelbooruTag {
    pub id: i64,
    pub name: String,
    pub count: i64,
    /// 0=general, 1=artist, 3=copyright, 4=character, 5=metadata.
    pub r#type: i64,
    pub ambiguous: bool,
}

#[derive(Debug, Clone, serde::Serialize)]
pub struct GelbooruTagSearchResponse {
    pub tags: Vec<GelbooruTag>,
    pub query: String,
}
